module/zfs/spa.c

```diff
@@ -957,24 +961,12 @@ spa_taskqs_init(spa_t *spa, zio_type_t t, zio_taskq_type_t q)
 	uint_t value = ztip->zti_value;
 	uint_t count = ztip->zti_count;
 	spa_taskqs_t *tqs = &spa->spa_zio_taskq[t][q];
-	uint_t flags = 0;
+	uint_t cpus, flags = 0;
```

Comment:
Switching from `ZTI_P` to  `ZTI_SCALE` caused us to accidentally drop the `TASKQ_DYNAMIC` flag which is currently only applied in the `ZTI_P` ( `ZTI_MODE_FIXED`) case.  I think it would make sense to just add `TASKQ_DYNAMIC` to the default `flags` here so it gets set for all of the `ZTI_MODE_*` cases.  This would mean  `TASKQ_DYNAMIC` would also be applied to the "write-issue" threads which uses `ZTI_BATCH` (`ZTI_MODE_BATCH`), but I think that would be desirable.  At least on Linux if we can avoid leaving the otherwise idle taskq threads around it's a usability win since they're all reported in the output of `ps -ef` , and the cost of re-spawning the threads as needed is negligible.

```suggestion
	uint_t cpus, flags = TASKQ_DYNAMIC;
```

Reply:
Considering how much this patch reduces the number of threads for average system, I am not sure it is so much needed.  But since you ask for it -- done.